root/components/EventsList.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {l} from '../static/scripts/common/i18n';
+import {lp_attributes} from '../static/scripts/common/i18n/attributes';
+import {withCatalystContext} from '../context';
+import loopParity from '../utility/loopParity';
+import ArtistRoles from '../static/scripts/common/components/ArtistRoles';
+import DescriptiveLink from '../static/scripts/common/components/DescriptiveLink';
+import EventLocations from '../static/scripts/common/components/EventLocations';
+import RatingStars from '../components/RatingStars';
+import SortableTableHeader from '../components/SortableTableHeader';
+import formatDatePeriod
+  from '../static/scripts/common/utility/formatDatePeriod';
+
+type Props = {|
+  +$c: CatalystContextT,
+  +artist?: ArtistT,
+  +artistRoles?: boolean,
+  +checkboxes?: string,
+  +events: $ReadOnlyArray<EventT>,
+  +noArtists?: boolean,
+  +noLocation?: boolean,
+  +noRatings?: boolean,
+  +noType?: boolean,
+  +order?: string,
+  +seriesItemNumbers?: $ReadOnlyArray<string>,
+  +sortable?: boolean,
+|};
+
+const EventsList = ({
+  $c,
+  artist,
+  artistRoles,
+  checkboxes,
+  events,
+  noArtists,
+  noLocation,
+  noRatings,
+  noType,
+  order,
+  seriesItemNumbers,
+  sortable,
+}: Props) => (
+  <table className="tbl">
+    <thead>
+      <tr>
+        {$c.user_exists && checkboxes ? (
+          <th style={{width: '1em'}}>
+            <input type="checkbox" />
+          </th>
+        ) : null}
+        {seriesItemNumbers ? <th style={{width: '1em'}}>{l('#')}</th> : null}
+        <th>
+          {sortable
+            ? (
+              <SortableTableHeader
+                label={l('Event')}
+                name="name"
+                order={order ? order : null}
+              />
+            )
+            : l('Event')}
+        </th>
+        {noType
+          ? null : (
```

Comment:
I'd suggest the same style used for `noType` below, so `{noType ? null : (` on one line, de-indenting the code inside the `()`, and closing with `)}`. It takes up less space and is a bit more readable, I think